src/openforms/formio/api/validators.py

```diff
@@ -108,31 +112,38 @@ def __call__(self, value: UploadedFile) -> None:
             raise serializers.ValidationError(
                 _("The provided file is not a {file_type}.").format(file_type=f".{ext}")
             )
-        # gh #4886
-        # We need to accept text/plain as a valid MIME type for CSV files.
-        # If the file does not strictly follow the conventions of CSV (e.g. non-standard delimiters),
-        # may not be considered as a valid CSV.
-        elif (
-            provided_mime_type == "text/csv"
-            and detected_mime_type == "text/plain"
-            and ext == "csv"
-        ):
-            return
-        elif detected_mime_type == "image/heic" and provided_mime_type in (
-            "image/heic",
-            "image/heif",
-        ):
-            return
-        # gh #4658
-        # Windows use application/x-zip-compressed as a mimetype for .zip files, which
-        # is deprecated but still we need to support it. Instead, the common case for
-        # zip files is application/zip or application/zip-compressed mimetype.
-        elif detected_mime_type == "application/zip" and provided_mime_type in (
-            "application/zip-compressed",
-            "application/x-zip-compressed",
-        ):
-            return
-        elif provided_mime_type != detected_mime_type:
+
+        # Handle edge cases where detection is not exact/reliable.
+        match (provided_mime_type, detected_mime_type, ext):
+            # gh #4886
+            # We need to accept text/plain as a valid MIME type for CSV files.
+            # If the file does not strictly follow the conventions of CSV (e.g.
+            # non-standard delimiters),
+            # may not be considered as a valid CSV.
+            case ("text/csv", "text/plain", "csv"):
+                return
+
+            # See earlier - heic/heif can be considered equivalent
+            case ("image/heic" | "image/heif", "image/heic", _):
+                return
+
+            # gh #4658
+            # Windows use application/x-zip-compressed as a mimetype for .zip files,
+            # which is deprecated but still we need to support it. Instead, the common
+            # case for zip files is application/zip or application/zip-compressed
+            # mimetype.
+            # libmagic 5.46+ doesn't detected application/zip anymore.
```

Comment:
```suggestion
            # libmagic 5.46+ doesn't detect application/zip anymore.
```
Looks much cleaner, just a small typo here :).

Reply:
fixed :)